src/components/posts/PostProvider.js

```diff
@@ -80,7 +80,7 @@ export const PostProvider = (props) => {
     }
 
     const getPostsByUser = (id) => {
-        return fetch(`http://localhost:8000/posts?user_id=${id}`, {
+        return fetch(`http://localhost:8000/posts?rareuser_id=${id}`, {
```

Comment:
`user_id` should not be used in the client. The only identifier that the client should have access to is the user's token.